pkg/ocm/kac.go

```diff
@@ -0,0 +1,258 @@
+package ocm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	kacv1 "github.com/stolostron/klusterlet-addon-controller/pkg/apis/agent/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// KACBuilder provides a struct for the KlusterletAddonConfig resource containing a connection to the cluster and the
+// KlusterletAddonConfig definition.
+type KACBuilder struct {
+	// Definition of the KlusterletAddonConfig used to create the object.
+	Definition *kacv1.KlusterletAddonConfig
+	// Object of the KlusterletAddonConfig as it is on the cluster.
+	Object *kacv1.KlusterletAddonConfig
+	// apiClient used to interact with the cluster.
+	apiClient runtimeclient.Client
+	// errorMsg used to store latest error message from functions that do not return errors.
+	errorMsg string
+}
+
+// NewKACBuilder creates a new instance of a KlusterletAddonConfig builder.
+func NewKACBuilder(apiClient *clients.Settings, name, nsname string) *KACBuilder {
```

Comment:
@klaskosk Can you verify if there aren't any required fields when creating a KlusterletAddonConfig? I see these on my cluster:
- applicationManager
- certPolicyController
- policyController
- searchCollector

Reply:
@trewest Yes, those are the only four fields [not listed as optional](https://pkg.go.dev/github.com/stolostron/klusterlet-addon-controller@v0.0.0-20240606130554-01338045271a/pkg/apis/agent/v1#KlusterletAddonConfigSpec), but [each field in that struct is optional](https://pkg.go.dev/github.com/stolostron/klusterlet-addon-controller@v0.0.0-20240606130554-01338045271a/pkg/apis/agent/v1#KlusterletAddonAgentConfigSpec) so I think it's okay to just leave them as their zero values